src/components/DistanceRequest.js

```diff
@@ -71,11 +71,12 @@ function DistanceRequest({transactionID, transaction, mapboxAccessToken}) {
     const hasRouteError = Boolean(lodashGet(transaction, 'errorFields.route'));
     const previousWaypoints = usePrevious(waypoints);
     const haveWaypointsChanged = !_.isEqual(previousWaypoints, waypoints);
-    const shouldFetchRoute = haveWaypointsChanged && !isOffline && !isLoadingRoute && TransactionUtils.validateWaypoints(waypoints);
+    const doesRouteExist = Boolean(lodashGet(transaction, 'routes.route0.geometry.coordinates'));
+    const shouldFetchRoute = (!doesRouteExist || haveWaypointsChanged) && !isLoadingRoute && TransactionUtils.validateWaypoints(waypoints);
 
     const waypointMarkers = _.filter(
         _.map(waypoints, (waypoint, key) => {
-            if (!waypoint || waypoint.lng === undefined || waypoint.lat === undefined) {
+            if (!waypoint || waypoint.lng === undefined || waypoint.lat === undefined || waypoint.lat === null || waypoint.lng === null) {
```

Comment:
NAB: Using `lodashHas` here might be better than the undefined and null checks too (which are just kinda ugly)